Objects/codeobject.c

```diff
@@ -2268,3 +2268,57 @@ _PyStaticCode_InternStrings(PyCodeObject *co)
     }
     return 0;
 }
+
+PyCodeObject *
+_Py_MakeTrampoline(const char *code, int codelen, int stacksize, const char *cname)
```

Comment:
So, this interface feels a bit strange to me. We only ever create one trampoline code object, right? Maybe take the `INTERPRETER_TRAMPOLINE_CODE` and all of the other arguments out of `pylifecycle.c` and move them here, since there's a pretty tight coupling between the arguments and the rest of the new code members.

Also, `codelen` should probably be the number of code units, not the number of bytes.

Reply:
Ah, maybe it's meant for reuse with future cleanup frames? In that case, it should be fine to just take a single opcode as an argument, right (`INTERPRETER_EXIT` in this case, maybe `CLEANUP_INIT` or `CLEANUP_SEND` in the future)?

Reply:
I expect to generalize it for specializing calls to classes, so the shim function can potentially do more work than that.
It makes sense to bundle the code, length, etc into some sort of struct.